### PR TITLE
Feature/defaults

### PR DESCRIPTION
**Pull request by evansde77**
Updates to allow for default settings file, plus a couple of bugfixes and hardening

### CODE REVIEW

File: src/dockerstache/context.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        and overlay the json file on top of that
+
+        """
+        if self._defaults_file is not None:
+            if not os.path.exists(self._defaults_file):
+                msg = "Unable to find defaults file: {}".format(self._defaults_file)
+                LOGGER.error(msg)
+                raise RuntimeError(msg)
+            with open(self._defaults_file, 'r') as handle:
+                self._defaults = json.load(handle)
```
Comment (ksnavely) [inline src/dockerstache/context.py:46]: Neat method

File: src/dockerstache/context.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+"""
+import os
+import json
+from . import get_logger
+
+
+LOGGER = get_logger()
+
+
+class Context(dict):
```
Comment (ksnavely) [inline src/dockerstache/context.py:17]: What's contextual about this class?

Comment (evansde77) [inline src/dockerstache/context.py:17]: Borrowed terminology from pystache: https://github.com/defunkt/pystache/blob/master/pystache/renderer.py#L421

**Comment by pengale** [comment]
Cool. LGTM/+1

**Pull request merged**